MantidQt/CustomInterfaces/src/Reflectometry/QtReflRunsTabView.cpp

```diff
@@ -72,15 +72,12 @@ void QtReflMainView::initLayout() {
   connect(qDataProcessorWidget,
           SIGNAL(comboProcessInstrument_currentIndexChanged(int)), this,
           SLOT(instrumentChanged(int)));
-  // Needed to Import/Export TBL, plot row and plot group
-  connect(qDataProcessorWidget,
-          SIGNAL(runAsPythonScript(const QString &, bool)), this,
-          SIGNAL(runAsPythonScript(const QString &, bool)));
 
-  m_presenter = boost::make_shared<ReflMainViewPresenter>(
-      this /*main view*/,
-      this /*currently this concrete view is also responsibile for prog reporting*/,
-      processorPresenter /*the table presenter*/);
+  // Create the presenter
+  m_presenter = std::make_shared<ReflRunsTabPresenter>(
+      this /* main view */,
+      this /* Currently this concrete view is also responsible for prog reporting */,
+      qDataProcessorWidget->getPresenter() /* The data processor presenter */);
```

Comment:
If you're not already, I would make sure that the pointers being passed back by `qDataProcessorWidget->getPresenter()` are of the form `Presenter * const`, this will prevent any client code from accidentally trying to call delete on a resource that is already being managed.